src/opencl-poh-verify/main.cpp

```diff
@@ -0,0 +1,277 @@
+#include "cl_common.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <pthread.h>
+
+#define USE_CLOCK_GETTIME
```

Comment:
One small change, it would be great if we can have this and also src/cuda-poh-verify/main.cu combined as src/poh-verify-test/main.cpp and then also put the `test_hashes_xx`/`test_num_arr` files in that directory as well so that it is clearer what they should be used with.

Reply:
I'm not sure if I understand correctly. but what you are saying is to make one binary that contains openCL and CUDA variants, I'm not sure if it can be done (but even if it can) architectures with other devices than NVIDIA will not work( or MAYBE it will work if they install CUDA environment and nvcc, and sometimes that is not possible).

About test files, I can move them in both home directories (opencl and CUDA).

Reply:
There is nothing that is cuda specific in `src/cuda-poh-verify/main.cu` and nothing OpenCL specific in ` src/opencl-poh-verify/main.cpp`. They are just generic CPU code. You can use just one source file, and then compile two executable binaries: `test_opencl_poh` which dynamically links with `libcl-crypt.so` and `test_cuda_poh` which dynamically links with `libcuda-crypt.so`.

Reply:
Oh, I got it now, it can be done :D. (I thought you were saying one binary with CUDA and OpenCL, but it was about object file).

Reply:
Done.